From 1f46deee93ba1fe5dab7d09f25476a89ec334749 Mon Sep 17 00:00:00 2001
From: Eric L <ewl+git@lavar.de>
Date: Fri, 5 May 2023 07:37:28 +0200
Subject: [PATCH] Fix cross-version issue with restrict_path

FIX: cross-version issue with 2.0.5 complaining about KeyError restrict_path, closes #872
---
 src/rdiff_backup/Globals.py                | 4 ++++
 src/rdiff_backup/Security.py               | 1 +
 tools/crossversion/playbook-smoke-test.yml | 2 +-
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/rdiff_backup/Globals.py b/src/rdiff_backup/Globals.py
index f9500bed6..82b8a8bc6 100644
--- a/src/rdiff_backup/Globals.py
+++ b/src/rdiff_backup/Globals.py
@@ -247,6 +247,10 @@
 # object.  Access is provided to increment error counts.
 ITRB = None
 
+# If this is set, it indicates that the remote connection should only
+# deal with paths inside of restrict_path.
+restrict_path = None  # compat200
+
 # If set, a file will be marked as changed if its inode changes.  See
 # the man page under --no-compare-inode for more information.
 compare_inode = 1
diff --git a/src/rdiff_backup/Security.py b/src/rdiff_backup/Security.py
index 701371bfd..a0f6d45f6 100644
--- a/src/rdiff_backup/Security.py
+++ b/src/rdiff_backup/Security.py
@@ -114,6 +114,7 @@ def reset_restrict_path(rp):
         "Function works locally not over '{conn}'.".format(conn=rp.conn))
     global _restrict_path, _restrict_path_list
     _restrict_path = rp.normalize().path
+    Globals.restrict_path = _restrict_path  # compat200
     _restrict_path_list = _restrict_path.split(b"/")
 
 
diff --git a/tools/crossversion/playbook-smoke-test.yml b/tools/crossversion/playbook-smoke-test.yml
index 199ed6d1a..44f965f86 100644
--- a/tools/crossversion/playbook-smoke-test.yml
+++ b/tools/crossversion/playbook-smoke-test.yml
@@ -63,7 +63,7 @@
   - name: check that the remote rdiff-backup works
     command: >
       rdiff-backup --remote-schema '{{ remote_schema }}' --test-server
-      {{ test_user }}\@{{ test_server }}::{{ remote_base_dir }}/simplebackup
+      {{ test_user }}\@{{ test_server }}::{{ remote_base_dir }}
     delegate_to: localhost
 
   - name: make a simple backup from the local directory to remote repo
